### ISSUE TITLE
Error: Guarantee failed: [ts->tv_nsec >= 0 && ts->tv_nsec < (1000LL * (1000LL * 1000LL))]

### ISSUE THREAD

**Issue by mbroadst** [rethinkdb/rethinkdb#4931]
I'm sorry I don't have time to research the cause of this atm, but wanted to put this here for reference. I've  just experienced a crash when switching nodes around in a cluster, resulting in the following:

```
2015-10-08T13:15:44.659193487 42377.176343s error: Error in src/time.cc at line 71:
2015-10-08T13:15:44.659304993 42377.176454s error: Guarantee failed: [ts->tv_nsec >= 0 && ts->tv_nsec < (1000LL * (1000LL * 1000LL))]
2015-10-08T13:15:44.659334099 42377.176483s error: Backtrace:
2015-10-08T13:15:44.994025935 42377.511179s error: Thu Oct  8 13:15:44 2015\n\n1: backtrace_t::backtrace_t() at ??:?\n2: format_backtrace(bool) at ??:?\n3: report_fatal_error(char const*, int, char const*, ...) at ??:?\n4: add_to_timespec(timespec*, int) at ??:?\n5: logs_artificial_table_backend_t::cfeed_machinery_t::run(peer_id_t const&, uuid_u const&, log_server_business_card_t const&, bool, auto_drainer_t::lock_t) at ??:?\n6: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (logs_artificial_table_backend_t::cfeed_machinery_t::*)(peer_id_t const&, uuid_u const&, log_server_business_card_t const&, bool, auto_drainer_t::lock_t)> (logs_artificial_table_backend_t::cfeed_machinery_t*, peer_id_t, uuid_u, log_server_business_card_t, bool, auto_drainer_t::lock_t)> >::run_action() at ??:?\n7: coro_t::run() at ??:?
2015-10-08T13:15:44.994200269 42377.511350s error: Exiting.
```

Hope that helps

**Comment by mbroadst** [comment]
other relevant information:

```
2015-10-08T01:29:27.763831874 0.280985s notice: Running rethinkdb 2.1.4~0trusty (GCC 4.8.2)...
2015-10-08T01:29:27.769366330 0.286519s notice: Running on Linux 3.16.0-50-generic x86_64
```

**Comment by danielmewes** [comment]
Thanks for reporting @mbroadst . Does this go away when you restart the server or is it a persistent issue?
If it's persistent, you can probably recover by deleting the log file from the rethinkdb data directory on the affected server.

Assigning @VeXocide. This appears to be originating from the logs table calling `add_to_timespec` with an invalid value.

The backtrace with newlines:
```
1: backtrace_t::backtrace_t() at ??:?
2: format_backtrace(bool) at ??:?
3: report_fatal_error(char const*, int, char const*, ...) at ??:?
4: add_to_timespec(timespec*, int) at ??:?
5: logs_artificial_table_backend_t::cfeed_machinery_t::run(peer_id_t const&, uuid_u const&, log_server_business_card_t const&, bool, auto_drainer_t::lock_t) at ??:?
6: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (logs_artificial_table_backend_t::cfeed_machinery_t::*)(peer_id_t const&, uuid_u const&, log_server_business_card_t const&, bool, auto_drainer_t::lock_t)> (logs_artificial_table_backend_t::cfeed_machinery_t*, peer_id_t, uuid_u, log_server_business_card_t, bool, auto_drainer_t::lock_t)> >::run_action() at ??:?
7: coro_t::run() at ??:?
```

**Comment by mbroadst** [comment]
@danielmewes it was not persistent, simply restarting the service resumed normal operation

**Comment by mbroadst** [comment]
@danielmewes this is now happening consistently

**Comment by danielmewes** [comment]
Have you tried deleting / moving out the current RethinkDB log files?
We're looking into the source of this error, but until then this might be a workaround.�

**Comment by mbroadst** [comment]
@danielmewes well it's not a persistent issue so no I haven't tried removing the log files. As it stands we can set in the unit file to respawn on crash, however I've been able to reproduce the follow scenario at least twice:
 
  - 3 nodes are joined together in a cluster with a primary replica/raft leader
  - data is inserted into a few tables on the first node
  - all tables are reconfigured to have 3 replicas as soon as the third node joins
  - first node goes down (the only one with actual data presently) because of this crash
  - nodes 2 and 3 seem to have received information about the tables but maybe not all the data, so they elect a new leader but no replicas are available for operations
  - node 1 comes back online after a restart, joins the cluster but is in some sort of completely inconsistent state including the fact that it just keeps timing out on a raft election and starting new ones (ad infinitum)

  TL;DR - once this crash happens in roughly those terms, the entire cluster must be brought down because nothing works

**Comment by VeXocide** [comment]
In CR 3285 by @danielmewes.

**Comment by VeXocide** [comment]
Merged into `next` and `v2.1.x` via commits fb9a2b9a8eda6cbbfc0716eac83caa2c1565b0b8 and a00733172c857d820a01f31b37b653c36b7319d0 respectively.

**Comment by mbroadst** [comment]
@VeXocide just confirmed on our side the crash is totally resolved, huzzah!
